text/src/autogluon/text/text_prediction/text_prediction.py

```diff
@@ -475,17 +475,20 @@ def fit(cls, train_data,
             scheduler_options = hyperparameters['hpo_params']['scheduler_options']
             if scheduler_options is None:
                 scheduler_options = dict()
+        scheduler_options['visualizer'] = visualizer
         if search_strategy.endswith('hyperband'):
             # Specific defaults for hyperband scheduling
             scheduler_options['reduction_factor'] = scheduler_options.get(
                 'reduction_factor', 4)
             scheduler_options['grace_period'] = scheduler_options.get(
                 'grace_period', 10)
             scheduler_options['max_t'] = scheduler_options.get(
-                'max_t', 10)
+                'max_t', 50)
 
         if recommended_resource['num_gpus'] == 0:
-            warnings.warn('Recommend to use GPU to run the TextPrediction task!')
+            raise RuntimeError('Only CPU is detected and we will not proceed to run TextPrediction.'
+                               ' GPU is required to run the model! You may set '
+                               '`ngpus_per_trial` to enable GPU.')
```

Comment:
May clarify: 
"we will not proceed to run TextPrediction"
->
"we will not fit TextPrediction models because they will train too slowly"

Reply:
Should there be a hyperparam flag that the user can specify if they really want to run TextPrediction on CPU?

Reply:
+1 on flag, but I recommend not adding it right now, probably better to wait until we have per-model cpu/gpu control.

Reply:
Haha, just added the environment variable support.